### ISSUE TITLE
Unity Editor crashes on starting Play mode in Editor

### ISSUE THREAD

**Issue by reekdeb** [resonance-audio/resonance-audio-unity-sdk#10]
I am experiencing numerous Unity Editor crashes when starting Play mode in editor. It happens when Resonance Audio is enabled and platform is either Desktop or Android. I have not tested on other platforms. Sometimes it works as expected sometimes it crashes the Editor.

Logs: [crash.dmp and error.log.zip](https://github.com/resonance-audio/resonance-audio-unity-sdk/files/1544416/crash.dmp.and.error.log.zip)

Here is `error.log` generated by Unity: 
```
Unity Editor [version: Unity 2017.2.0f3_46dda1414e51]

audiopluginresonanceaudio.dll caused an Access Violation (0xc0000005)
  in module audiopluginresonanceaudio.dll at 0033:dfabfd60.

Error occurred at 2017-12-05_121610.
I:\ProgramFiles\GameDev\Unity\Editor\Unity.exe, run by Reekdeb.
78% memory in use.
4096 MB physical memory [880 MB free].
8960 MB paging file [3249 MB free].
134217728 MB user address space [134215090 MB free].
Read from location 6ab213e8 caused an access violation.

...
```

**Comment by mauskopf** [comment]
Thanks for reporting this issue and sharing the crash.dmp and error.log. We are looking into the cause and we'll update this thread shortly!

**Comment by jkammerl** [comment]
Thanks for the report. Could you some more information when the crash happens? Does it happen during initialization, shutdown or randomly during runtime?

**Comment by reekdeb** [comment]
During initialization, just after clicking Play button in Unity Editor. Perhaps the first frame is rendered, then hangs the editor just for a second, the Editor crashes and generates bug report. This same crash happens in every other cases and makes it completely unusable.

**Comment by anokta** [comment]
Hi Reekdeb,

We can't seem to reproduce the crash on our end using Unity 2017.2.0f3. Would you mind sharing more details on the reproduction steps?

Particularly, it would really help us to know:
- Is the crash observable with a new vanilla Unity project, using the unmodified ResonanceAudioDemo scene that comes with the SDK? If not, is there a scene setup/steps that you can share that leads to a crash? e.g. adding rooms/Ambisonic soundfields which triggers the issue?
- Are there any other assets/SDKs in the Unity project apart from Resonance Audio Unity SDK (e.g., Google VR Unity SDK)? If so, does the issue still observable without the addition of such assets?

It would also be helpful if you could send out a simple Unity project that you can reproduce the crash - if possible.

Thanks

**Comment by anokta** [comment]
In the meantime, one thing you can try out is to comment out the line [ResonanceAudioListener.cs#L74](https://github.com/resonance-audio/resonance-audio-unity-sdk/blob/master/Assets/ResonanceAudio/Scripts/ResonanceAudioListener.cs#L74) and see if the crash still occurs:
`
// ResonanceAudio.UpdateAudioListener(this);
`

Alternatively, you can also try to remove the `ResonanceAudioListener` component from your scene completely to see if that helps (you may need to restart Unity after removing the component & saving the scene to verify).

**Comment by reekdeb** [comment]
@anokta I commented out `ResonanceAudio.UpdateAudioListener(this);` and removed `ResonanceAudioListener` component. In both cases crash still occurs frequently.

I created a new test project. Crashes are less frequent but still occurs. Roughly 2 crashes in 30 times when entering Play mode.

Here is the test project including crash dumps, error logs and a screencast of how to let it crash: [ResonanceAudio-1.0.0-TestCrash.7z](https://drive.google.com/open?id=1gmkmMIeieZ7DDTZeJu2OwFQd8CpEdBjD)

**Comment by anokta** [comment]
Thanks for providing the test project! 

This issue should be fixed in [v1.1.0](https://github.com/resonance-audio/resonance-audio-unity-sdk/releases/tag/v1.1.0). Before we close out the issue, could you please verify if the issue is resolved on your end with the updated version?

**Comment by reekdeb** [comment]
Now it is crashing every time I hit the Play button. Even demo scenes which are included in SDK are crashing the Editor upon hitting Play button. All my three projects are crashing the Editor same way.

Unity 2017.2.1f0
Resonance Audio Unity SDK 1.1.0. 

Crash Logs: [Crash_Logs_U2017.2.1f0_RA1.1.0.zip](https://github.com/resonance-audio/resonance-audio-unity-sdk/files/1564517/Crash_Logs_U2017.2.1f0_RA1.1.0.zip)

**Comment by vedmaka** [comment]
+1 crashing on 2017.3.0f2 after several seconds after Play was hit in editor, no sound or anything, using 1.1.0 sdk package

**Comment by anokta** [comment]
Thanks for the feedback! Both issues should be fixed now in [v1.1.1](https://github.com/resonance-audio/resonance-audio-unity-sdk/releases/tag/v1.1.1).

**Comment by reekdeb** [comment]
Thanks to all. The issue is fixed now.
